File: Assignment12/Program2.c

//Problem Statement :Write a program which accept number from user and return the count of odd digits

#include<stdio.h>

int CountOdd(int iNo)
{
    auto int iDigit = 0;
    auto int iCount = 0;

      
    if(iNo < 0)
    {
        iNo = -iNo;
    }

    while(iNo > 0)
    {
        iDigit = iNo % 10;

        if(iDigit % 2 != 0)  //Logic to find odd digit in number 
        {
            iCount++;  // count++ to store the odd Numbers 
        }
        iNo = iNo / 10;
    }

   return iCount;        //Return count to caller function to print the result on screen
}


int main()
{
   auto int iValue = 0;
   auto int iRet = 0;

   printf("Enter a number :");
   scanf("%d" , &iValue);

   iRet = CountOdd(iValue);

   printf("%d",iRet);

    return 0;
}
//Test  Step
/* 
 Input : 1018
 Output: 2

 Input :-1018
 Output :2

 Input ;8462
 Output :0

*/

File: Assignment34/Program2.java


//1.Write a java Program which accept number from user and return the count of Odd Digits
/*
//Input 2395
//Output 3
 
Input 1018
Output 3

Input -1018
Output 2

Input 8462
Output 0

*/
import java.util.*;


class Program2
{
    public static int CountOdd(int iNo)
    {
        int iDigit = 0;
        int iCount = 0;
          
        while(iNo != 0)
        {
            iDigit = iNo % 10;
            if((iDigit % 2) != 0)
            {
                iCount++;
            }
            iNo = iNo / 10;
            
           
        }
        return iCount;
    }
    

    public static void main(String Args[])
    {
        Scanner sobj = new Scanner(System.in);
        int iValue = 0, iRet = 0;

        System.out.println("Enter Number :");
        iValue = sobj.nextInt();
        
        iRet = CountOdd(iValue);
        System.out.println("Odd Count is : "+iRet);
    }
}
